app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala

```diff
@@ -116,6 +116,26 @@ object TransactionQueries extends StrictLogging {
         getKnownTransactionAction(txHash, blockHash, timestamp, gasAmount, gasPrice).map(Some.apply)
     }
 
+  def getOutputRefTransactionAction(key: Hash)(
+      implicit ec: ExecutionContext): DBActionR[Option[Transaction]] = {
+    sql"""
+    SELECT tx_hash
+    FROM outputs
+    WHERE main_chain = true AND key = $key
+    """.as[Transaction.Hash].flatMap { txHashes =>
+      txHashes.headOption match {
```

Comment:
We'd better introduce a utility function for unique result query, so it will log errors when there are more than one query result